h2o-webserver-iface/src/main/java/water/webserver/iface/H2OHttpConfig.java

```diff
@@ -0,0 +1,46 @@
+package water.webserver.iface;
+
+import java.util.Map;
+
+/**
+ * Holds configuration relevant to HTTP server.
+ */
+public class H2OHttpConfig {
```

Comment:
This config never changes. A simple look at this class raises few questions:

1. Are all the properties really required ?
2. Can those properties ever change ? If yes, then which ones ?

Wouldn't it be better to introduce a way to force developers fill the required arguments ? I assume port and IP are mandatory, but jks-related properties are optional. There are several ways to achieve this. 

- Builder pattern, with builder's constructor forcing to supply mandatory args.
- Constructor of this class requiring the mandatory args. Mandatory fields set to final. 

Builders looks better to me, but requires more work. 

This is something I'd personally do, but I don't see it as a necessity in order to finish this PR. Just something to consider.

Reply:
I like the idea, will think about what's best here, to balance complexity and usefulness.
And yes, I just checked all the fields and some can be removed now, after some refactors happened.
Thanks for pointing this out.

Reply:
second thoughts:

- developers (implementing `iface` module) are never exposed to this, only the core is doing it
- I significantly minimized the number of config params, now there is only 8, so constructor might address your concern, if it persists; but I am not very convinced that it is overall better solution, so I will leave it as is for now, perhaps you come with more opinions